### PR TITLE
Fix filesystem linking

### LINKED ISSUE #423: Cannot build F3D 1.3.1 Flatpak
**Describe the bug**
I was trying to update F3D to 1.3.1 but encounter an error.

```
[ 85%] Built target libf3d
[ 87%] Generating F3DIcon.h, F3DIcon.cxx
[ 91%] Building CXX object application/CMakeFiles/f3d.dir/F3DIcon.cxx.o
[ 93%] Building CXX object application/CMakeFiles/f3d.dir/F3DOptionsParser.cxx.o
[ 93%] Building CXX object application/CMakeFiles/f3d.dir/F3DConfigFileTools.cxx.o
[ 97%] Building CXX object application/CMakeFiles/f3d.dir/F3DStarter.cxx.o
[ 97%] Building CXX object application/CMakeFiles/f3d.dir/main.cxx.o
[100%] Linking CXX executable ../bin/f3d
/usr/lib/gcc/x86_64-unknown-linux-gnu/11.3.0/../../../../x86_64-unknown-linux-gnu/bin/ld: cannot find -lstdc++fs
collect2: error: ld returned 1 exit status
make[2]: *** [application/CMakeFiles/f3d.dir/build.make:170: bin/f3d] Error 1
make[1]: *** [CMakeFiles/Makefile2:448: application/CMakeFiles/f3d.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```
I look into it a bit, it seems like it need g++. But g++ is already available 
```
g++ (GCC) 11.3.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**To Reproduce**
Full error log
https://buildbot.flathub.org/#/builders/35/builds/13200/steps/6/logs/stdio

Flatpak PR
https://github.com/flathub/io.github.f3d_app.f3d/pull/9#issuecomment-1243091779

**System Information:**
Operating System: openSUSE Tumbleweed 20220907
KDE Plasma Version: 5.25.4
KDE Frameworks Version: 5.97.0
Qt Version: 5.15.5
Kernel Version: 5.19.7-1-default (64-bit)
Graphics Platform: X11
Processors: 12 × AMD Ryzen 5 4600H with Radeon Graphics
Memory: 7.6 GiB of RAM
Graphics Processor: NVIDIA GeForce GTX 1650 Ti/PCIe/SSE2
Manufacturer: LENOVO
Product Name: 82B5
System Version: Lenovo Legion 5 15ARH05
flatpak-builder --version
flatpak-builder 1.2.2

**F3D Information**
Building from source 1.3.1

Do you have some clue on to this?
Thanks.

### PR DESCRIPTION
**Pull request by Meakk**
Fix #423

### CODE REVIEW

**Comment by Meakk** [comment]
More complex than what I expected. What matters is the libstdc++ version, not the GCC version.  
For example in our CI, we are using clang 10 with libstdc++ 8, so we actually need the link option to fs.

File: library/testing/CMakeLists.txt [hunk 1]
```diff
@@ -112,9 +112,5 @@ if(Qt5_FOUND)
   target_link_libraries(libf3dSDKTests Qt5::OpenGL)
 endif()
 
-if (UNIX AND NOT APPLE)
-  target_link_libraries(libf3dSDKTests stdc++fs)
```
Comment (mwestphal) [inline library/testing/CMakeLists.txt:116]: why not needed in the lib ?

Comment (Meakk) [reply library/testing/CMakeLists.txt:116]: It's the testing executable, and we remove support of C++17 from it.

Comment (mwestphal) [reply library/testing/CMakeLists.txt:116]: right

File: application/CMakeLists.txt [hunk 2]
```diff
@@ -29,9 +29,13 @@ endif()
 add_executable(f3d ${F3D_SOURCE_FILES})
 target_link_libraries(f3d PUBLIC libf3d)
 
-if (UNIX AND NOT APPLE)
-  target_link_libraries(f3d PRIVATE stdc++fs)
-endif ()
+if(UNIX AND NOT APPLE)
+  option(F3D_APPLICATION_LINK_FILESYSTEM "Link with libstdc++fs" ON)
```
Comment (mwestphal) [inline application/CMakeLists.txt:33]: Add a point about this in the troubleshooting section ?

File: README.md [hunk 3]
```diff
@@ -490,6 +490,10 @@ Be sure that VTK has been built with *OpenImageDenoise* support (`VTKOSPRAY_ENAB
   * If only a few format have working thumbnails, then it is an issue with mime types
   * If no formats have working thumbnails, then it is an issue with the f3d.thumbnailer file
 
+> I have a link error related to `stdc++fs` not found.
+
+With some C++ STD library version, explicit linking to `stdc++fs` is not required. We provide a CMake option `F3D_APPLICATION_LINK_FILESYSTEM` that you can set to `OFF` to workaround this issue.
+
```
Comment (mwestphal) [inline README.md:496]: Is not supported maybe ?

**Review by mwestphal — APPROVED** [review]
Approved with a comment if you want

**Pull request merged**